### PR TITLE
TB 389:Add capability to generate and enrich default import map with links to external apps

### PR DESCRIPTION
**Pull request by ItsMurumba**
Add capability to allow for the importmap to be enriched with new links for all external apps that do not exist in the monorepo but are already registered within openhim core.

The PR introduces an [endpoint](/importmaps) used to fetch transformed/enriched import map file

## Summary by CodeRabbit

- **New Features**
  - Introduced a new endpoint for ImportMaps to retrieve and transform app data.
- **Configuration**
  - Added `openhimConsoleBaseUrl` setting to various configuration files.
- **Infrastructure**
  - Updated `docker-compose.yml` with new environment variables for authentication and OpenHIM console settings.
- **Tests**
  - Enhanced integration tests to cover new ImportMaps functionality.

### CODE REVIEW

**Review by drizzentic — APPROVED** [review]

File: infrastructure/docker-compose.yml [hunk 1]
```diff
@@ -14,12 +14,19 @@ services:
 
   openhim-core:
     container_name: openhim-core
-    image: jembi/openhim-core:latest
+    # image: jembi/openhim-core:latest
```
Comment (drizzentic) [inline infrastructure/docker-compose.yml:17]:
```suggestion
    image: jembi/openhim-core:latest

```

File: infrastructure/docker-compose.yml [hunk 2]
```diff
@@ -14,12 +14,19 @@ services:
...
+    image: openhim-core:latest
     restart: unless-stopped
     environment:
-      mongo_url: "mongodb://mongo-db/openhim-development"
-      mongo_atnaUrl: "mongodb://mongo-db/openhim-development"
-      NODE_ENV: "development"
+      - mongo_url="mongodb://mongo-db/openhim-development"
+      - mongo_atnaUrl="mongodb://mongo-db/openhim-development"
+      # - NODE_ENV="development"
```
Comment (drizzentic) [inline infrastructure/docker-compose.yml:23]:
```suggestion
      - NODE_ENV="development"

```

File: src/constants.js [hunk 3]
```diff
@@ -15,3 +17,22 @@ export const JWT_PATTERN =
 
 export const CUSTOM_TOKEN_PATTERN =
   /^ *(?:[Cc][Uu][Ss][Tt][Oo][Mm]) +([A-Za-z0-9\-._~+/]+=*) *$/
+
+const OPENHIM_CONSOLE_BASE_URL = config.openhimConsoleBaseUrl || 'http://localhost:9000'
```
Comment (drizzentic) [inline src/constants.js:21]:
```suggestion
const OPENHIM_CONSOLE_BASE_URL = config.get('openhimConsoleBaseUrl')
```

File: infrastructure/docker-compose.yml [hunk 4]
```diff
@@ -14,12 +14,19 @@ services:
...
+      - api_authenticationTypes=["token", "basic", "openid", "local"]
+      - authentication_enableCustomTokenAuthentication=true
+      - authentication_enableJWTAuthentication=true
+      - authentication_jwt_secretOrPublicKey=secret
+      - authentication_jwt_algorithms=HS256
+      - authentication_jwt_issuer=openhim
```
Comment (drizzentic) [inline infrastructure/docker-compose.yml:29]:
```suggestion
      - authentication_jwt_issuer=openhim
      - openhimConsoleBaseUrl= "http://localhost:9000"
```

File: test/integration/appsAPITests.js [hunk 5]
```diff
@@ -261,5 +264,46 @@ describe('API Integration Tests', () => {
...
+        const importMaps = [{ name: 'map1', url: 'url1' }, { name: 'map2', url: 'url2' }]
+        const ctx = { request: { query: {} }, body: {}, status: 0 }
+    
+        AppModelAPI.find.resolves(importMaps)
+
+        const expectedImports = {
+          ...DEFAULT_IMPORT_MAP_PATHS,
+          map1: 'url1',
+          map2: 'url2'
+        }
```
Comment (drizzentic) [inline test/integration/appsAPITests.js:287]: Is this constant used anywhere?

Comment (ItsMurumba) [reply test/integration/appsAPITests.js]: No. This is now removed

File: src/model/apps.js [hunk 6]
```diff
@@ -32,3 +34,21 @@ const AppSchema = new Schema({
...
+
+    try {
+      const response = await axios.get('/importmaps', {})
+      loggers.info('Importmaps API request successful')
+
+    } catch (e) {
+      loggers.error('Error making Importmaps API request', e);
+    }
+  })
+}
```
Comment (bradsawadye) [inline src/model/apps.js:52]: Clean up is necessary here

Comment (ItsMurumba) [reply src/model/apps.js]: Done

**Review by drizzentic — DISMISSED** [review]

**Pull request closed without merging**